src/search.js

```diff
@@ -0,0 +1,47 @@
+const checkMatch = (value, searchTerm) => value.toLowerCase().includes(searchTerm);
+
+const filter = (currentNode, searchTerm, onlyPublic) => {
+  let newChildren = [];
+  let isPublic = true;
+  let result = false;
+  let hasChildren;
+
+  currentNode.hasOwnProperty("children") && currentNode.children.forEach(child => {
+    let keepChild = filter(child, searchTerm, onlyPublic);
```

Comment:
This is doing depth-first search recursively; in a refactor I would change it to iterative to avoid stack overflow.